File: app/src/main/java/org/michaelbel/tjgram/data/entity/Likes.java
package org.michaelbel.tjgram.data.entity;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.HashMap;

public class Likes implements Serializable {
    @Expose @SerializedName("count") public int count;
    @Expose @SerializedName("is_hidden") public boolean isHidden;
    @Expose @SerializedName("is_liked") public int isLiked;
    @Expose @SerializedName("likers") public HashMap<String, Liker> likersMap;
    @Expose @SerializedName("summ") public int summ;

    public Likes(int count, boolean isHidden, int isLiked, int summ) {
        this.count = count;
        this.isHidden = isHidden;
        this.isLiked = isLiked;
        this.summ = summ;
    }

    public Likes(int summ) {
        this.summ = summ;
    }
}

File: app/src/main/java/org/michaelbel/tjgram/data/entities/Likes.kt
package org.michaelbel.tjgram.data.entities

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.*

data class Likes (
    @Expose @SerializedName("count") var count: Int = 0,
    @Expose @SerializedName("is_hidden") var isHidden: Boolean = false,
    @Expose @SerializedName("is_liked") var isLiked: Int = 0,
    @Expose @SerializedName("likers") var likersMap: HashMap<String, Liker>? = null,
    @Expose @SerializedName("summ") var summ: Int = 0
): Serializable